src/Actions/RedirectIfTwoFactorAuthenticatable.php

```diff
@@ -76,15 +75,17 @@ protected function validateCredentials($request)
             });
         }
 
-        $model = $this->guard->getProvider()->getModel();
+        $user = $this->guard->getProvider()->retrieveByCredentials(
+            $request->only(Fortify::username(), 'password')
+        );
 
-        return tap($model::where(Fortify::username(), $request->{Fortify::username()})->first(), function ($user) use ($request) {
-            if (! $user || ! Hash::check($request->password, $user->password)) {
-                $this->fireFailedEvent($request, $user);
+        if (! $user) {
+            $this->fireFailedEvent($request, $user);
 
-                $this->throwFailedAuthenticationException($request);
-            }
-        });
+            $this->throwFailedAuthenticationException($request);
+        }
+
+        return $user;
```

Comment:
The verification might be not enough. `retrieveByCredentials()` does not expect password matches, only an email address does. Password hash validation is done by `validateCredentials()`, but its call was not included in the PR.

I think it is the cause of the PR reverted. 

@taylorotwell Is it right?